src/libraries/Microsoft.Extensions.Configuration.Xml/src/XmlStreamConfigurationProvider.cs

```diff
@@ -129,81 +184,264 @@ public override void Load(Stream stream)
-        }
-
         private static string GetLineInfo(XmlReader reader)
         {
             var lineInfo = reader as IXmlLineInfo;
             return lineInfo == null ? string.Empty :
                 SR.Format(SR.Msg_LineInfo, lineInfo.LineNumber, lineInfo.LinePosition);
         }
 
-        private static void ProcessAttributes(XmlReader reader, Stack<string> prefixStack, IDictionary<string, string> data,
-            Action<XmlReader, Stack<string>, IDictionary<string, string>, XmlWriter> act, XmlWriter writer = null)
+        private static void ProcessAttributes(XmlReader reader, XmlConfigurationElement element)
         {
+            if (reader.AttributeCount > 0)
+            {
+                element.Attributes = new List<XmlConfigurationElementAttributeValue>();
+            }
+
+            var lineInfo = reader as IXmlLineInfo;
+
             for (int i = 0; i < reader.AttributeCount; i++)
             {
                 reader.MoveToAttribute(i);
 
+                var lineNumber = lineInfo?.LineNumber;
+                var linePosition = lineInfo?.LinePosition;
+
                 // If there is a namespace attached to current attribute
                 if (!string.IsNullOrEmpty(reader.NamespaceURI))
                 {
                     throw new FormatException(SR.Format(SR.Error_NamespaceIsNotSupported, GetLineInfo(reader)));
                 }
 
-                act(reader, prefixStack, data, writer);
+                element.Attributes.Add(new XmlConfigurationElementAttributeValue(reader.LocalName, reader.Value, lineNumber, linePosition));
             }
 
             // Go back to the element containing the attributes we just processed
             reader.MoveToElement();
         }
 
         // The special attribute "Name" only contributes to prefix
-        // This method adds a prefix if current node in reader represents a "Name" attribute
-        private static void AddNamePrefix(XmlReader reader, Stack<string> prefixStack,
-            IDictionary<string, string> data, XmlWriter writer)
+        // This method retrieves the Name of the element, if the attribute is present
+        // Unfortunately XmlReader.GetAttribute cannot be used, as it does not support looking for attributes in a case insensitive manner
+        private static string GetName(XmlReader reader)
         {
-            if (!string.Equals(reader.LocalName, NameAttributeKey, StringComparison.OrdinalIgnoreCase))
+            string name = null;
+
+            while (reader.MoveToNextAttribute())
             {
-                return;
+                if (string.Equals(reader.LocalName, NameAttributeKey, StringComparison.OrdinalIgnoreCase))
+                {
+                    // If there is a namespace attached to current attribute
+                    if (!string.IsNullOrEmpty(reader.NamespaceURI))
+                    {
+                        throw new FormatException(SR.Format(SR.Error_NamespaceIsNotSupported, GetLineInfo(reader)));
```

Comment:
Why do we need to throw this exception here? It seems like it in `ProcessAttributes`.

Reply:
I think - please forgive me if I'm wrong - you are confusing two methods here:

1. `private static void ProcessAttributes(XmlReader reader, XmlConfigurationElement element)`
2. `void ProcessElementAttributes(Prefix prefix, XmlConfigurationElement element)`

The first method is invoked during the parsing stage. It reads the XML attributes from the XML document and adds instances of `XmlConfigurationElementAttributeValue` to the current `XmlConfigurationElement`

The second method is invoked during the processing stage. It produces configuration key values for each attribute.

To avoid future confusion, I have renamed `ProcessAttributes` (the first method) to `ReadAttributes`.

If I misunderstood you and it was not a matter of confusion, can you please clarify why you think that exception does not belong there?